neqo-transport/src/server.rs

```diff
@@ -434,47 +458,42 @@ impl Server {
     fn accept_connection(
         &mut self,
         attempt_key: AttemptKey,
-        dcid: ConnectionId,
-        orig_dcid: Option<ConnectionId>,
+        initial: InitialDetails,
         dgram: Datagram,
+        orig_dcid: Option<ConnectionId>,
         now: Instant,
-        quic_version: QuicVersion,
     ) -> Option<Datagram> {
         qinfo!([self], "Accept connection {:?}", attempt_key);
         // The internal connection ID manager that we use is not used directly.
         // Instead, wrap it so that we can save connection IDs.
 
-        let local_initial_source_cid = self.cid_manager.borrow_mut().generate_cid();
-
         let cid_mgr = Rc::new(RefCell::new(ServerConnectionIdManager {
             c: None,
             cid_manager: self.cid_manager.clone(),
             connections: self.connections.clone(),
+            saved_cids: Vec::new(),
         }));
 
         let sconn = Connection::new_server(
             &self.certs,
             &self.protocols,
             &self.anti_replay,
-            self.cid_manager.clone(),
-            quic_version,
-            local_initial_source_cid.clone(),
+            cid_mgr.clone(),
```

Comment:
The bug was here.  I didn't pass in the wrapper.  It was the underlying manager that was being passed.

That meant that any new connection IDs we might generate wouldn't be registered properly.